src/utils.js

```diff
@@ -126,7 +126,10 @@ const moduleRegExp = /\.module(s)?\.\w+$/i;
 const icssRegExp = /\.icss\.\w+$/i;
 
 function getModulesOptions(rawOptions, loaderContext) {
-  const { resourcePath } = loaderContext;
+  const resourcePath =
+    // eslint-disable-next-line no-underscore-dangle
+    loaderContext._module.matchResource || loaderContext.resourcePath;
```

Comment:
woohoo! should this also be the logic for local ident generation?

Reply:
Yep, you are right, otherwise we will get the same class names.

After fix local ident generation should we merge it and release? All is fixed?

Reply:
i think so, seems like this covers it all. hard to know for sure until i try it out on a code base. Thanks again for helping me out here

Reply:
Yep, here interesting case, when we have: 
```
import three from './button.module.scss!=!./base64-loader?LmZvbyB7IGNvbG9yOiByZWQ7IH0=!./simple.js';
import four from './other.module.scss!=!./base64-loader?LmZvbyB7IGNvbG9yOiByZWQ7IH0=!./simple.js';
```

We run the loader only once, why? Because ES modules executed only once. So I don't think here bug. But we can solve it using:
```
// i.e. adding `?` with different values
import three from './button.module.scss!=!./base64-loader?LmZvbyB7IGNvbG9yOiByZWQ7IH0=!./simple.js?foo=bar';
import four from './other.module.scss!=!./base64-loader?LmZvbyB7IGNvbG9yOiByZWQ7IH0=!./simple.js?foo=baz';
```

Reply:
ah interesting, so the query param acts like a cache buster. That solution seems fine to me

Reply:
Yep, you can generate it and please avoid using `Math.random()`, because cache will be broken

Reply:
yeah in my case it'd be easy to use a stable identifier here. This wouldn't solve the className hash generation issue tho right? I'm guessing you discovered this righting a test for that?

Reply:
Solved, please check code

Reply:
arg, i'm trying this out now and `_module.matchResource` is null locally. Going to debug a bit to see why will report back

Reply:
If you don't have `matchResource` in request, it will be nullish

Reply:
yeah this is for requests with it in it. For some reason i am seeing the module get created twice, one with the matchResource and one without, the one without is what makes it to the css-loader. It maybe has to do with ESM, still trying to figure it out